### ISSUE TITLE
Problem defining an empty multidimensional array

### ISSUE THREAD

**Issue by Ivo-Balbaert** [vlang/v#2981]
**V version:** 0.1.23
**OS:** 
Windows 10 and Linux

**What did you do?**
Tried to compile the following program:
```v
fn main() {
	m2arr := [][]int 
}
```

**What did you expect to see?**
program compiled normally

 
**What did you see instead?**
```
warning: test.v:2:6: `m2arr` declared and not used
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c: In function 'main__main':
C:\Users\CVO\AppData\Local\Temp\v\test.tmp.c:3340:50: error: expected expression before ')' token
 array_ m2arr=new_array_from_c...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen. 
```

---
Same happens for other types instead of int.

**Comment by hillerstorm** [comment]
Yep, the only way I have found so far to define a two-dimensional array is like this:
```v
mut arr := [[]int]
arr.delete(0)
```

which is... far from satisfying 😄

**Comment by Ivo-Balbaert** [comment]
arr := [[]int]
println(arr)

This notation works indeed! I am willing to accept this notation if this is going to be V's syntax.
If Alex confirms this, we can close this issue.

**Comment by medvednikov** [comment]
Interesting.

It should be `arr := [][]int`, but `arr := [[]int]` is valid V.

I'll have to think about it.

**Comment by medvednikov** [comment]
I think I'll make only `arr := [][]int` valid.

**Comment by borgespires** [comment]
`arr := [[]int]` is valid. However, it creates an array with one element `[[]]` instead of an empty array of arrays `[]`.

**Comment by medvednikov** [comment]
Good point.

**Comment by Ivo-Balbaert** [comment]
The top example now gives a V error:

test.v:2:14: error: expr(): bad token `]` 
    1 | fn main() {
    2 |     m2arr := [][]int
      |                 ^
    3 |     println(m2arr)
    4 | }

**Comment by Ivo-Balbaert** [comment]
Fixed in 0.1.28 with:

m2arr := [][]int{}
